Sources/View/Widget/ExpiryDatePickerView.swift

```diff
@@ -19,79 +19,91 @@
+    }()
+
+    private lazy var locale: Locale = {
+        Locale(identifier: Locale.preferredLanguages[0])
+    }()
+
+    private lazy var calendar: Calendar = {
+        var calendar = Calendar.current
+        calendar.locale = locale
+        return calendar
+    }()
+
+    private lazy var localizedYearDateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.locale = locale
+        formatter.setLocalizedDateFormatFromTemplate("y")
+        return formatter
+    }()
+
+    private lazy var expiryDateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "expiry_date_picker_format".localized()
+        return formatter
+    }()
+
+    private var selectedDateComponents = DateComponents()
+    private var localizedMonths = [String]()
+    private var localizedYears = [String]()
 
     convenience init(value: String?, frame: CGRect = .zero) {
         self.init(frame: frame)
-
-        if let values = value?.components(separatedBy: "-"),
-            values.count == 2,
-            let defaultMonth = Int(values[1]),
-            let defaultYear = Int(values[0]) {
-            month = defaultMonth
-            year = defaultYear
-        } else {
-            let currentDate = Date()
-            let currentCalendar = Calendar.current
-            month = currentCalendar.component(.month, from: currentDate)
-            year = currentCalendar.component(.year, from: currentDate)
+        setSelectedDateComponents(for: currentDate)
+        if let value = value, let date = expiryDateFormatter.date(from: value) {
+            setSelectedDateComponents(for: date)
         }
-
-        setupMonths()
-        setupYears(add: 10)
+        setupMonthsYears()
 
         delegate = self
         dataSource = self
 
-        // pick current month as default month for the picker and place the selected month in the center of picker
         selectRow(month - 1, inComponent: 0, animated: false)
+        selectRow(year % firstYear, inComponent: 1, animated: false)
     }
 
-    // MARK: UIPicker Delegate / Data Source
-    func numberOfComponents(in pickerView: UIPickerView) -> Int {
-        return 2
+    func setSelectedDateComponents(for date: Date) {
+        selectedDateComponents.calendar = calendar
+        selectedDateComponents.setValue(1, for: .day)
+        selectedDateComponents.setValue(calendar.component(.month, from: date), for: .month)
+        selectedDateComponents.setValue(calendar.component(.year, from: date), for: .year)
     }
 
+    // MARK: UIPicker Delegate / Data Source
+    func numberOfComponents(in pickerView: UIPickerView) -> Int { return 2 }
+
     func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
-        month = selectedRow(inComponent: 0) + 1
-        year = years[selectedRow(inComponent: 1)]
+        selectedDateComponents.setValue(
+            component == 0 ? row + 1 : row + firstYear,
```

Comment:
Could you move these desitions to some variables?  because you are doing 3 operations in one line. 
Suggestion: 
```
let value = component == 0 ? row + 1 : row + firstYear
let type = component == 0 ? .month : .year
selectedDateComponents.setValue(value, for: type)
```

Reply:
no problem, fixed